src/main/java/com/minecolonies/coremod/tileentities/TileEntityBarrel.java

```diff
@@ -0,0 +1,240 @@
+package com.minecolonies.coremod.tileentities;
+
+import com.minecolonies.api.configuration.Configurations;
+import com.minecolonies.api.util.ItemStackUtils;
+import com.minecolonies.coremod.items.ModItems;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Items;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.NetworkManager;
+import net.minecraft.network.play.server.SPacketUpdateTileEntity;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumParticleTypes;
+import net.minecraft.util.ITickable;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.world.World;
+import net.minecraft.world.WorldServer;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Random;
+
+public class TileEntityBarrel extends TileEntity implements ITickable
+{
+    private boolean                                 done           = false;
+    private int                                     items          = 0;
+    private int                                     timer          = 0;
```

Comment:
jdoc for public methods and for all fields please.
Comments please  in jdoc and not with // on next line